### PR TITLE
Plumb prng through as an arg where needed instead of making it a clas…

### LINKED ISSUE #49: Make prng an arg to Distribution::sample instead of a member of Distribution

### PR DESCRIPTION
**Pull request by emilyfertig**
…s member.

I considered keeping prng as a member of (H)IRM, but plumbing it through added minimal code complexity and it seems better to pass it in where it's used instead of have it hidden in the class.

Closes #49

### CODE REVIEW

File: cxx/emissions/bitflip.hh [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
 #include <cassert>
+
 #include "emissions/base.hh"
 
 // A *deterministic* Emission class that always emits not(clean).
 // Most users will want to combine this with Sometimes.
 class BitFlip : public Emission<bool> {
  public:
-  BitFlip() {};
+  BitFlip(){};
```
Comment (ThomasColthurst) [inline cxx/emissions/bitflip.hh:11]: Missing space between () and {}.

File: cxx/emissions/simple_string.hh [hunk 2]
```diff
@@ -18,9 +19,7 @@ class SimpleStringEmission : public Emission<std::string> {
   BetaBernoulli insertions;
   BetaBernoulli deletions;
 
-  SimpleStringEmission() : substitutions(nullptr),
-                           insertions(nullptr),
-                           deletions(nullptr) {};
+  SimpleStringEmission(){};
```
Comment (ThomasColthurst) [inline cxx/emissions/simple_string.hh:22]: Same.

**Review by ThomasColthurst — APPROVED** [review]
THANK YOU THANK YOU THANK YOU THANK YOU!

File: cxx/emissions/bitflip.hh (hunk 1 above)
Comment (emilyfertig) [reply cxx/emissions/bitflip.hh]: Done (+elsewhere).

**Comment by emilyfertig** [comment]
No prob! I needed something mechanical to do during the ChiExpert session on Friday...

**Pull request merged**